src/Notification/NotificationServiceProvider.php

```diff
@@ -49,29 +65,22 @@ protected function setNotificationTypes()
             new ConfigureNotificationTypes($blueprints)
         );
 
-        foreach ($blueprints as $blueprint => $channelsEnabledByDefault) {
-            $this->addType($blueprint, $channelsEnabledByDefault);
+        foreach ($blueprints as $blueprint => $driversEnabledByDefault) {
+            $this->addType($blueprint, $driversEnabledByDefault);
         }
     }
 
-    protected function addType(string $blueprint, array $channelsEnabledByDefault)
+    protected function addType(string $blueprint, array $driversEnabledByDefault)
     {
         Notification::setSubjectModel(
             $type = $blueprint::getType(),
             $blueprint::getSubjectModel()
         );
 
-        User::addPreference(
-            User::getNotificationPreferenceKey($type, 'alert'),
-            'boolval',
-            in_array('alert', $channelsEnabledByDefault)
-        );
-
-        if ((new ReflectionClass($blueprint))->implementsInterface(MailableInterface::class)) {
-            User::addPreference(
-                User::getNotificationPreferenceKey($type, 'email'),
-                'boolval',
-                in_array('email', $channelsEnabledByDefault)
+        foreach (NotificationSyncer::getNotificationDrivers() as $driverName => $driver) {
+            $driver->registerType(
+                $blueprint,
+                in_array($driverName, $driversEnabledByDefault)
```

Comment:
One thing that came up as I was working on my PWA extension: it would be preferable to pass in the array of drivers enabled by default here (either instead of, or in addition to, argument 2). That gives extensions a bit more flexibility (for example, I enable PWA by default if email is enabled by default).

Reply:
Very well